### PR TITLE
List support: clear method 

### PR DESCRIPTION
**Pull request by mustapha-belbiad**
Add support for the pop() method of Python lists to the semantic stage.
Add handling of that method to the Python printer. This fixes : #1691.

### CODE REVIEW

File: pyccel/ast/builtin_methods/list_methods.py [hunk 1]
```diff
@@ -14,7 +14,10 @@
 from pyccel.ast.internals import PyccelInternalFunction
 
 
-__all__ = ('ListAppend','ListPop')
+__all__ = ('ListAppend',
+           'ListPop',
+           'ListClear'
+           )
```
Comment (EmilyBourne) [inline pyccel/ast/builtin_methods/list_methods.py:20]: Please fix the `pyccel_lint` warning

**Review by EmilyBourne — CHANGES_REQUESTED** [review]
Just one minor comment

**Review by EmilyBourne — APPROVED** [review]
Looks good to me

File: tests/epyccel/test_epyccel_lists.py [hunk 2]
```diff
@@ -117,3 +117,40 @@ def f():
...
+
+def test_clear_1(language):
+
+    def clear_1():
+        a = [1, 2, 3]
+        a.clear()
+        return a
+
+    epyc_last_element = epyccel(clear_1, language = language)
+    pyccel_result = epyc_last_element()
```
Comment (yguclu) [inline tests/epyccel/test_epyccel_lists.py:130]: Why do you use the name `epyc_last_element` for the pyccelized function? That does not make much sense to me. I suggest `epyc_clear_1`.

The same comment applies the following two test functions.

**Review by yguclu — CHANGES_REQUESTED** [review]
Looks good. I left a few minor comments.

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,7 @@ All notable changes to this project will be documented in this file.
 -   #1529 : Add Python support for list method `append()`
 -   #1690 : Add support for `pop()` method of list class
 -   #1575 : Add support for homogeneous tuple type annotations on variables.
+-   #1691 : Add support for `clear()` method of list class
```
[the remark this answers is not in the document]
Comment (mustapha-belbiad) [reply CHANGELOG.md]: fixed : 5a50b55

File: pyccel/codegen/printing/pycode.py [hunk 4]
```diff
@@ -1148,6 +1148,10 @@ def _print_ListPop(self, expr):
         name = self._print(expr.list_variable)
         return f"{name}.pop({args})"
 
+    def _print_ListClear(self, expr):
+        name = self._print(expr.list_variable)
+        return f"{name}.clear()\n"
+
```
[the remark this answers is not in the document]
Comment (mustapha-belbiad) [reply pyccel/codegen/printing/pycode.py]: fixed : 4e4b91c

File: tests/epyccel/test_epyccel_lists.py (hunk 2 above)
Comment (mustapha-belbiad) [reply tests/epyccel/test_epyccel_lists.py]: fixed : b64b4e5

**Review by yguclu — APPROVED** [review]

**Pull request merged**